dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -256,83 +264,102 @@ public Workflow getWorkflow(String repositoryId, Optional<Workflow> existingWork
-                if (workflowVersion.getReference().equals(branch)) {
-                    filePath = workflowVersion.getWorkflowPath();
-                    sourceFiles = workflowVersion.getSourceFiles();
-                    version = workflowVersion;
-                }
+            Workflow workflow = (Workflow)entry;
+            workflow.getWorkflowVersions().forEach(workflowVersion -> {
+                String filePath = workflowVersion.getWorkflowPath();
+                updateVersionMetadata(filePath, workflowVersion, type, repositoryId);
+            });
+        }
+        return entry;
+    }
+
+    /**
+     * Sets the default version if there isn't already one present.
+     * This is required because entry-level metadata depends on the default version
+     *
+     * @param entry
+     * @param repositoryId
+     */
+    public void setDefaultBranch(Entry entry, String repositoryId) {
+        if (entry.getDefaultVersion() == null) {
+            String branch = getMainBranch(entry, repositoryId);
+            if (branch == null) {
+                String message = String.format("%s : Error getting the main branch.", repositoryId);
+                LOG.info(message);
+            } else {
+                Set<Version> workflowVersions = entry.getWorkflowVersions();
+                Optional<Version> firstWorkflowVersion = workflowVersions.stream()
+                        .filter(workflowVersion -> {
+                            String reference = workflowVersion.getReference();
+                            return reference != null && reference.equals(branch);
+                        }).findFirst();
+                firstWorkflowVersion.ifPresent(version -> entry.checkAndSetDefaultVersion(version.getName()));
             }
         }
+    }
 
+    private void updateVersionMetadata(String filePath, Version version, DescriptorLanguage type, String repositoryId) {
+        Set<SourceFile> sourceFiles = version.getSourceFiles();
+        String branch = version.getName();
         if (Strings.isNullOrEmpty(filePath)) {
-            LOG.info(repositoryId + " : No descriptor found for " + branch + ".");
-            return entry;
+            String message = String.format("%s : No descriptor found for %s.", repositoryId, branch);
+            LOG.info(message);
         }
-
         if (sourceFiles == null || sourceFiles.isEmpty()) {
-            LOG.info(repositoryId + " : Error getting descriptor for " + branch + " with path " + filePath);
-            return entry;
+            String message = String.format("%s : Error getting descriptor for %s with path %s", repositoryId, branch, filePath);
+            LOG.info(message);
+            if (version.getReference() != null) {
+                String readmeContent = getREADMEContent(repositoryId, version.getReference());
+                if (readmeContent != null && !readmeContent.isBlank()) {
```

Comment:
You could optionally use `StringUtils.isNotBlank()` instead, which will handle the null check.